keps/sig-windows/1122-windows-csi-support/README.md

```diff
@@ -768,6 +777,348 @@ Downgrading the version of csi-proxy.exe to one that is not supported by all ins
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+  If csi-proxy binary is deployed as a binary or windows service during node start up time, enabling this feature will require reprovisioning of a node.
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+It depends on cloud provider implementation. For example in GCE, environment variable ENABLE_CSI_PROXY is implemented to disable/enable CSI-proxy installation.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+-->
+No
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+It is up to cloud provider to have their own plan. For example, GKE will use version control to rollout, upgrade and rollback this feature.
+Please also see (#upgrade--downgrade-strategy)
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+It is up to the cloud provider to tie CSI-proxy version to cluster version. When clusters rollout and rollback, the CSI-proxy should be udpated accordingly.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+N/A
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+N/A
```

Comment:
What about csi-proxy binary version upgrades/downgrades?

Reply:
Different provider might use different way to upgrade/downgrade. For GKE, this is be controlled by setting environment variable to control csi-proxy version which can tied to GKE cluster version.

Reply:
Yes, but should we at least do some basic upgrade/downgrade testing (could be manual) so that we know the basic mechanics will work and can provide guidance to distros on the proper/supported ordering?

Reply:
we could provide manual test guidance using pd csi driver. Will it work?

Reply:
Yes some development documentation on how you could test upgrades/downgrades would be helpful. We can also investigate if the existing kubernetes statefulset upgrade test can also be leveraged for Windows.

Reply:
I think we could have a section about the steps we discussed to upgrade the client first (pd-csi) and then the server (csi-proxy binary) independently.